pkg/block/blocktest/adapter.go

```diff
@@ -217,11 +218,20 @@ func testAdapterMultipartUpload(t *testing.T, adapter block.Adapter, storageName
 	}
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
+			blockstoreType := adapter.BlockstoreType()
 			obj := block.ObjectPointer{
 				StorageNamespace: storageNamespace,
 				Identifier:       c.path,
 				IdentifierType:   block.IdentifierTypeRelative,
 			}
+			// List parts on non-existing part
+			_, err := adapter.ListParts(ctx, obj, "invalidId", block.ListPartsOpts{})
+			if blockstoreType != block.BlockstoreTypeS3 {
```

Comment:
This is beautiful: it will force us to fix the test as soon as we fix Azure / GCS! :-)